### ISSUE TITLE
1/3th of the map the game slows down to much to make this work

### ISSUE THREAD

**Issue by mmuziek** [the1killer/sm_overview_ahk#5]
Sadly this way of making the map doesnt work for me
i have a beast of a Gaming pc with a RTX 2060 and Intel Core I7 9700

the game works fine but then after 4 rows the stutter starts.
in the end after row 5 the game lags behind and the map doesnt generate like its suppose to anymore.

see attached screenshot i ended the render as i saw it failing
![Screenshot_2](https://user-images.githubusercontent.com/1270969/98385186-ff6fff80-204e-11eb-9027-df5c419eb91c.png)

**Comment by Captnwalker1** [comment]
How much RAM do you have? I noticed after a few runs of generating the map my games memory usage is 2-4x what it was when I started, so seems there may be a memory leak.

You could always try adding another teleport + sleep to help make sure an area loads in.

I developed this on a 2080 on a i7 7820x, but seems more dependent on ram/disk than CPU to load the tiles. Do you have the game on an SSD?

**Comment by the1killer** [comment]
How much RAM do you have? I noticed after a few runs of generating the map my games memory usage is 2-4x what it was when I started, so seems there may be a memory leak.

You could always try adding another teleport + sleep to help make sure an area loads in.

I developed this on a 2080 on a i7 7820x, but seems more dependent on ram/disk than CPU to load the tiles. Do you have the game on an SSD?

**Comment by mmuziek** [comment]
well i have 32GB ram on my pc so that should not be the issue :)
the game itself is on a HDD but my windows is on SSD i will try it again tomorrow with more teleports?

**Comment by mmuziek** [comment]
well
that didnt work. i changed the delay to be 5000 + added a teleport and delay of 3000 but this is the result:
https://mcmap.bitcubes.eu/scrap/
as u can see the map is almost half when i see the game starting to stutter
from that point the game slows down with every new tp resulting in 0FPS for 10 seconds later in the  mapgen causing duplicate sections like visible. 
after 7 good lines the 8th one starts to glitch already and then the next are getting worse by the line.
may i ask if  there is a wat to retake missing points if this was in it. i could delete the broken images and start the system again to retake the missing ones :)
it would be as easy as check if file exists otherwise go to ..... and take screenshot and redo...

**Comment by the1killer** [comment]
Yeah i've been meaning to look at doing a regenerate type thing, just haven't gotten around to it. Sadly havent been playing as much SM lately.

I will try and look into this to see if my game is running any worse now with newer updates. But thats another question, have you tried turning down everything except render distance?

**Comment by mmuziek** [comment]
Yes all off or low only render distance to max

**Comment by mmuziek** [comment]
Well i have altered the script to make it resumeable or fill in the missing pictures:
```
Loop, %yloop% {
        
        ; Mouse Move to prevent screensaver
        MouseMove, 20, 20, 2, R
        x := startx
        ix := origx
        teleport(x,y)
        Sleep, 3000
        Loop, %xloop% {
            File2 := dirname "\" ix "," iy ".jpg"
            if (!FileExist(File2)){
            teleport(x,y)
            Sleep, 4000 ; Change this delay if your machine takes a while to load the tiles after teleporting
            teleport(x,y)
            Sleep, 4000 ; Change this delay if your machine takes a while to load the tiles after teleporting
            teleport(x,y)
            Send {Alt Down}
            Sleep 50
            Send Z
            Sleep 50
            Send {Alt Up}
            Sleep, 300
            takeScreenshot(ix,iy)
            Sleep, 500
            Send {Alt Down}
            Sleep 50
            Send Z
            Sleep 150
            Send {Alt Up}
            Sleep 50
            }
            ; Sleep, 500
            x := x + jump
            ix := ix + 1
            
        }
        y := y + jump
        iy := iy - 1
    }
```

**Comment by the1killer** [comment]
@mmuziek sorry for the delay, loaded up a fresh map and tried it and I am getting similar results, quite the stutter, so sadly I think its related to the game. I will attempt to open a bug with them but not even sure where to start with that. I will add your existing image check into the ahk file and add a feature flag for it (off by default).

**Comment by gamingoninsulin** [comment]
I was reading trough and tought will is help turn on vertical sync ? Maybe it helps

**Comment by the1killer** [comment]
@mmuziek did want to mention I published the tile based method that is exported from the game just by loading the save, making it WAYYYYyyyyyy faster. Slightly not as beautiful as this screenshot method but better overall, and probably all I will be working on going forward.

[https://github.com/the1killer/sm_overview](https://github.com/the1killer/sm_overview)
